File: modules/chat-core/src/main/java/com/shterneregen/securelan/chat/discovery/PeerDiscoveryListener.java
package com.shterneregen.securelan.chat.discovery;

public interface PeerDiscoveryListener {
    void onPeerDiscovered(DiscoveredPeer peer);

    void onPeerExpired(DiscoveredPeer peer);

    void onDiscoveryError(String message, Throwable cause);
}

File: modules/chat-core/src/main/kotlin/com/shterneregen/securelan/chat/discovery/PeerDiscoveryListener.kt
package com.shterneregen.securelan.chat.discovery

interface PeerDiscoveryListener {
    fun onPeerDiscovered(peer: DiscoveredPeer)

    fun onPeerExpired(peer: DiscoveredPeer)

    fun onDiscoveryError(message: String, cause: Throwable)
}
